gadget/mkfs.go

```diff
@@ -33,18 +33,26 @@ import (
 // filesystem label, and populates it with the contents of provided root
 // directory.
 func MkfsExt4(img, label, contentsRootDir string) error {
+	return mkfsExt4(img, label, contentsRootDir)
+}
+
+// mkfsExt4WithoutMetadataChecksum also creates an EXT4 filesystem in given image
+// file, but using checksum protection for block group descriptors instead of
+// metadata checksum.
+func MkfsExt4WithoutMetadataChecksum(img, label, contentsRootDir string) error {
+	// disable metadata checksum, which were unsupported in Ubuntu 16.04 and
+	// Ubuntu Core 16 systems and would lead to a boot failure if enabled
```

Comment:
Maybe we should reference the original discussion from https://github.com/snapcore/snapd/pull/6997#discussion_r293967140